src/Aspire.Hosting/Extensions/ResourceExtensions.cs

```diff
@@ -118,6 +118,16 @@ public static bool TryGetContainerImageName(this IResource resource, [NotNullWhe
         return false;
     }
 
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="resource"></param>
+    /// <returns></returns>
+    public static IEnumerable<IResource> GetDependencies(this IResource resource)
```

Comment:
Should this be transitive?

Reply:
No, I think the graph walk should be another API. For e.g. we wouldn't want to write the closure to the manifest, but we would walk it when walking the graph to do something in process.